src/containers/Network/NodesTable.tsx

```diff
@@ -99,59 +106,200 @@ export const NodesTable: FC<{ nodes: NodeData[] }> = ({
+        {node.quorum}
+      </td>
+      <td className={getClassNamesFor('load-factor right', 'load')}>
         {node.load_factor && node.load_factor > 1
           ? node.load_factor.toFixed(2)
           : ''}
       </td>
-      <td className="latency right">
+      <td className={getClassNamesFor('latency right', 'latency')}>
         {node.io_latency_ms && node.io_latency_ms > 1}
       </td>
     </tr>
   )
 
+  const compareSemanticVersions = (
+    a: string,
+    b: string,
+    returnValue: number,
+  ) => {
+    const a1 = a.split('.')
+    const b1 = b.split('.')
+
+    const len = Math.min(a1.length, b1.length)
+
+    for (let i = 0; i < len; i++) {
+      const a2 = +a1[i] || 0
+      const b2 = +b1[i] || 0
+
+      if (a2 !== b2) {
+        return a2 > b2 ? returnValue : returnValue * -1
+      }
+    }
+    return b1.length - a1.length
+  }
+
+  if (nodes !== null) {
+    const sort = (key: any, order: string) => {
+      const returnValue = order === 'desc' ? 1 : -1
+
+      if (key === 'peers') {
+        nodes.sort((a, b) =>
+          a.inbound_count + a.outbound_count >
+          b.inbound_count + b.outbound_count
+            ? returnValue * -1
+            : returnValue,
+        )
+      } else if (key === 'ledger_history') {
+        nodes.sort((a, b) =>
+          getLedgerHistory(a.ledgers, ledgerRange).count >
+          getLedgerHistory(b.ledgers, ledgerRange).count
+            ? returnValue * -1
+            : returnValue,
+        )
+      } else if (key === 'rippled_version') {
+        nodes.sort((a, b) =>
+          compareSemanticVersions(a.version, b.version, returnValue),
+        )
+      } else {
+        nodes.sort((a, b) => (a[key] > b[key] ? returnValue * -1 : returnValue))
+      }
+    }
+
+    sort(sortedField, sortOrder)
+  }
+
+  const getClassNamesFor = (name, order) =>
+    sortedField === order ? `${name} sorted` : name
+  const getClassArrowFor = (field) => {
+    if (sortedField === field && sortOrder === 'desc') {
+      return 'arrow down'
+    }
+    if (sortedField === field && sortOrder === 'asc') {
+      return 'arrow up'
+    }
+    return ''
+  }
+
   const content = nodes ? (
     <table className="basic">
       <thead>
         <tr>
-          <th className="pubkey">{t('node_pubkey')}</th>
-          <th className="ip">{t('ip')}</th>
-          <th className="server-state center">{t('state')}</th>
-          <th className="version">{t('rippled_version')}</th>
-          <th className="last-ledger">{t('last_ledger')}</th>
-          <th className="uptime">{t('uptime')}</th>
-          <th className="peers right">{t('peers')}</th>
+          <th className={getClassNamesFor('pubkey', 'pubkey')}>
+            <a href="#" onClick={() => requestSort('pubkey')}>
+              <i className={getClassArrowFor('pubkey')} />
+              {t('node_pubkey')}
+            </a>
+          </th>
+          <th className={getClassNamesFor('ip', 'ip')}>
```

Comment:
Maybe we can create a reusable component like `NodeTableHeader` and just call `<NodeTableHeader title='ip'>` for instance